src/mon/OSDMonitor.cc

```diff
@@ -1273,6 +1288,18 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     t->erase(OSD_METADATA_PREFIX, stringify(*p));
   pending_metadata.clear();
   pending_metadata_rm.clear();
+
+  // and pg creating, also!
+  creating_pgs_t pending_creatings = creating_pgs;
+  for (auto old_pool : pending_inc.old_pools) {
+    pending_creatings.created_pools.erase(old_pool);
+  }
+  scan_for_creating_pgs(osdmap.get_pools(), &pending_creatings);
+  // do not scan the pending_inc.new_pools now, as the mapping_job will not
+  // update the mapping with the new osdmap until on_active() after refreshing
```

Comment:
it seems like we could keep the creating_pgs list strictly in sync with the osdmap, since it doesn't include any mappings.  it's only the notifications that need crush calculations and may need a delay?